index.js

```diff
@@ -6,6 +6,22 @@ function decode(buffer) {
   let tag = buffer.readUInt8(bytesRead);
   bytesRead += 1;
 
+  const cls = tag >> 6;
+  const form = tag >> 5 & 1;
+  let tagCode = tag & 0b11111;
+
+  if (tagCode === 0b11111) {
+    let longTag = '';
+
+    do {
+      tagCode = buffer.readUInt8(bytesRead);
+      bytesRead += 1;
+      longTag += (tagCode & 0x7f).toString(16);
+    } while (tagCode & 0b10000000);
```

Comment:
Nitpick: It would be clearer to use `& 0x80` here for symmetry with the `& 0x7f` above.